benchmarks/url_benchmarks/url_resolve_flat_i18n_off/benchmark.py

```diff
@@ -0,0 +1,25 @@
+from ...utils import bench_setup
+from django.urls import Resolver404, resolve
+from django.conf import settings
+
+class UrlResolve:
+
+    def setup(self):
+        bench_setup()
+        settings.ROOT_URL_CONF = "url_resolve_flat_i18n_off.urls"
```

Comment:
https://docs.djangoproject.com/en/4.0/topics/settings/#altering-settings-at-runtime

I'm not sure we should be over riding settings? This would be a permanent change rather than just this benchmark? 🤔

If we need to override settings then maybe one of these methods could be used?


https://docs.djangoproject.com/en/4.0/topics/testing/tools/#overriding-settings-1

Reply:
Since django.setup() was begin called every time I thought that the settings in settings.py would be applied and the changes made in one benchmark would not affect the other. Today I will check if the settings get overridden and if they do I will use the method mentioned.

Reply:
`django.setup()` is in a `try` block and we're catching a `RuntimeError`. I think that's the error that gets raised if you try and call `django.setup()` more than once. 

Interested to hear your findings.

Reply:
To avoid modifying settings in specific benchmarks, I refactored the benchmarks such that each benchmark now has its settings.py file, in this [branch](https://github.com/deepakdinesh1123/django-asv/tree/temporary). The benchmarks are also much cleaner this way, if this format seems okay I will refactor all the benchmarks in this manner.

Reply:
I'm not sure this will work.  `django.setup()` can only be called once. Therefore I would expect if we are running multiple benchmarks then django will be setup with the first settings file only and will be ignored for the following benchmarks.

Reply:
I looked into this a bit more and I found that even calling `django.setup()` or `bench_setup()` twice in the same file is not generating an error and I also removed the try except to see if `RuntimeError` was being raised but I did not get any errors,  so I tried to change settings using the method that you mentioned but the benchmark suite cannot be built when the benchmark class inherits `Testcase` class so I had to create another class and benchmark like this 

class UrlResolveTest(TestCase):

    def test_i8n_off(self):
        with self.settings(ROOT_URL_CONF='url_resolve_flat_i18n_off.urls', USE_I8N = False):
            . . .
           . . .  

  class UrlResolve:

    def setup(self):
        bench_setup()

    def time_i8n_off(self):
        UrlResolveTest().test_i8n_off()

The performance of the benchmark was similar to when the benchmark  was in bench_urls.py. Should I follow this method to change settings?

Reply:
The benchmarks still pass even if this line is removed 🤔

Reply:
The URLs in the file [urls.py](https://github.com/smithdc1/django-asv/blob/main/benchmarks/url_benchmarks/url_resolve_flat/urls.py) of `url_resolve_flat` and the ones in `url_resolve_flat_i8n_off` were the same, since the URLs of `url_resolve_flat` was added to the URLs of the project [here](https://github.com/smithdc1/django-asv/blob/a5d33e24cad47ac20bc5497d71636c1f4ab4d1f0/benchmarks/urls.py#L11) the benchmarks passed even when the line was removed.